web/CASTNXT/app/javascript/utils/RangeFilter.js

```diff
@@ -0,0 +1,139 @@
+    };
+    const handleLowerFilterChange = (event) => {
+      const newLowerBound = event.target.value;
+      updateFilterValue(Number(newLowerBound), filterValueState[1]);
+    };
+  
+    return (
+      <Box
+        sx={{
+          display: 'inline-flex',
+          flexDirection: 'row',
+          alignItems: 'end',
+          height: 48,
+          pl: '20px',
+        }}
+      >
+        <TextField
+          name="lower-bound-input"
+          placeholder="From"
+          label="From"
+          variant="standard"
+          value={Number(filterValueState[0])}
+          onChange={handleLowerFilterChange}
+          type="number"
+          inputRef={focusElementRef}
+          sx={{ mr: 2 }}
+        />
+        <TextField
+          name="upper-bound-input"
+          placeholder="To"
+          label="To"
+          variant="standard"
+          value={Number(filterValueState[1])}
+          onChange={handleUpperFilterChange}
+          type="number"
+          //InputProps={applying ? { endAdornment: <SyncIcon /> } : {}}
+        />
+      </Box>
+    );
+  }
+  
+InputNumberInterval.propTypes = {
+    applyValue: PropTypes.func.isRequired,
+    focusElementRef: PropTypes.oneOfType([
+      PropTypes.func,
+      PropTypes.shape({
+        current: PropTypes.any.isRequired,
+      }),
+    ]),
+    item: PropTypes.shape({
+      /**
+       * The column from which we want to filter the rows.
+       */
+      columnField: PropTypes.string.isRequired,
+      /**
+       * Must be unique.
+       * Only useful when the model contains several items.
+       */
+      id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+      /**
+       * The name of the operator we want to apply.
+       * Will become required on `@mui/x-data-grid@6.X`.
+       */
+      operatorValue: PropTypes.string,
+      /**
+       * The filtering value.
+       * The operator filtering function will decide for each row if the row values is correct compared to this value.
+       */
+      value: PropTypes.any,
+    }).isRequired,
+};
+
+export const extendedNumberOperators = [
+    ...getGridNumericColumnOperators(),
+    //This feature is available in mui-v5. Data-grid needs to be migrated.
+    /*{
+      label: 'Between',
+      value: 'between',
+      getApplyFilterFn: (filterItem) => {
+        if (!Array.isArray(filterItem.value) || filterItem.value.length !== 2) {
+          return null;
+        }
+        if (filterItem.value[0] == null || filterItem.value[1] == null) {
+          return null;
+        }
+        console.log('here');
+        return ({ value }) => {
+          console.log(value);
+          return (
+            value !== null &&
+            filterItem.value[0] <= value &&
+            value <= filterItem.value[1]
+          );
+        };
+      },
+      InputComponent: InputNumberInterval,
+      //InputComponentProps: { type: 'number' },
+    },*/
+];
```

Comment:
Why is this logic commented out? If it is not needed, can you remove it ?

Reply:
Rohan, I have asked @mounikabalivada312 to migrate to Material UI 5.0 and test once again. It should ideally work. If it doesn't we will remove this piece at that point of time.

Reply:
Ok ! Looks good to me.